.github/workflows/cherry-pick.yml

```diff
@@ -2,126 +2,104 @@
+
+jobs:
+  # Scans the PR for cherry pick comments, in addition to the default cherry pick target
+  collect_target_branches:
+    if: github.event.pull_request.merged == true
     runs-on: ubuntu-latest
-    permissions:
-      contents: read
-      pull-requests: write
+    name: Collect target branches from Pull Request
     outputs:
-      status: ${{ steps.pr_info.outputs.status }}
-      base_ref: ${{ steps.pr_info.outputs.base_ref }}
-      merge_commit: ${{ steps.pr_info.outputs.merge_commit }}
+      cherry_pick_target_branches: ${{ steps.target_branches.outputs.cherry_pick_target_branches }}
     env:
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
-      - name: Extract PR information
-        id: pr_info
-        run: |
-          echo "status=$(gh pr view ${{ github.event.issue.number }} --repo ${{ github.repository }} --json state --jq '.state')" >> $GITHUB_OUTPUT
-          echo "base_ref=$(gh pr view ${{ github.event.issue.number }} --repo ${{ github.repository }} --json baseRefName --jq '.baseRefName')" >> $GITHUB_OUTPUT
-          echo "merge_commit=$(gh pr view ${{ github.event.issue.number }} --repo ${{ github.repository }} --json mergeCommit --jq '.mergeCommit.oid')" >> $GITHUB_OUTPUT
-
-      - name: Check that PR satisfies requirements for cherry-pick action and notify if not
+      # Collect target branches in a list. The list includes:
+      # - default target branch
+      # - any branch specified in PR comments with the syntax
+      #     /cherry-pick <branch-1> <branch-2>
+      # The final list will only contain unique branches
+      - name: Create list of target branches
+        id: target_branches
         run: |
-          if [[ "${{ steps.pr_info.outputs.status }}" != "MERGED" ]] || [[ "${{ steps.pr_info.outputs.base_ref }}" != fasttrack/* ]]; then
-            gh pr comment ${{ github.event.issue.number }} \
-              --repo ${{ github.repository }} \
-              --body 'Cherry pick action is only enabled for merged pull requests to `fasttrack/*` branches'
-          fi
+          cherry_pick_branches_json=$(gh pr view ${{ github.event.number }} \
+                                       --repo ${{ github.repository }} \
+                                       --json comments \
+                                       --jq '.comments
+                                              | map(select(.authorAssociation == "MEMBER" or .authorAssociation == "OWNER")
+                                                    | .body
+                                                    | select(startswith("/cherry-pick "))
+                                                    | ltrimstr("/cherry-pick ")
+                                                    | split(" "))
+                                              | flatten
+                                              | . += ["${{ fromJSON(env.CHERRY_PICK_BRANCH_MAPPING)[github.base_ref] }}"]
+                                              | unique')
```

Comment:
This is somewhat complex - could you add a comment explaining what it does? Ideally with a sample input and output.

Reply:
Added some explanation in the comment along with examples of how the json object transforms